### ISSUE TITLE
Sass Compilation Module Error

### ISSUE THREAD

**Issue by kaelri** [JeffreyWay/laravel-mix#3378]
-   Laravel Mix Version: 6.0.49
-   Node Version: v21.0.0
-   NPM Version: 10.2.0
-   OS: macOS Sonoma 14.0

### Description:

I suddenly started getting compilation errors on a project that was fine a few days ago. See below for a representative error message.

```shell
❯ npx mix             

✖ Mix
  Compiled with some errors in 396.75ms

1 WARNING in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)

ERROR in ./style.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleError: Module Error (from ./node_modules/sass-loader/dist/cjs.js):
Cannot read properties of undefined (reading 'indexOf')
    at Object.emitError (/Users/kaelri/Downloads/lmtest/node_modules/webpack/lib/NormalModule.js:614:6)
    at getSassImplementation (/Users/kaelri/Downloads/lmtest/node_modules/sass-loader/dist/utils.js:63:21)
    at Object.loader (/Users/kaelri/Downloads/lmtest/node_modules/sass-loader/dist/index.js:29:59)

2 ERRORS in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
webpack compiled with 3 errors and 1 warning
```

### Steps To Reproduce:

I put together a super-simple demo that you can clone and run:

- https://github.com/kaelri/laravel-mix-error-demo 
- `npm install && npm run build`

In short, my package includes only Laravel Mix, plus the dependencies that LM automatically installs as of today:

```json
{
	"devDependencies": {
		"laravel-mix": "^6.0.49",
		"resolve-url-loader": "^5.0.0",
		"sass": "^1.69.4",
		"sass-loader": "^12.6.0"
	},
	"scripts": {
		"build": "npx mix"
	}
}
```

Then a barebones webpack config:

```js
const mix = require('laravel-mix');
mix.sass( 'style.scss', 'style.css' );
```

And finally, some vanilla Sass. No includes, mixins, or anything else fragile:

```css
body {
	background: yellow;
}
```

This fails with the error copied above. I checked both webpack and sass-loader for recent issues that might be related, but no hits. Happy to provide any more information or test results that may be helpful!

**Comment by kurorido** [comment]
downgrade the node version to 18 solved the problem temporarily.

**Comment by olivM** [comment]
> downgrade the node version to 18 solved the problem temporarily.

it didn't for me :(
i tried multiple 20.x versions and even 21, but still the same error

**Comment by olivM** [comment]
it's working on a ubuntu 22.04 with node 20.6
but the error occurs on macOS 14 with node 20.6

i'll try to test it on macOS 13

**Comment by olivM** [comment]
can't reproduce the issue on ventura 13.4.1 node 20.5

so it seems to be Sonoma related, i got the 23A344 build

**Comment by olivM** [comment]
get it to work on macOS 14, thanks to [bun](https://bun.sh/)

**Comment by kaelri** [comment]
> get it to work on macOS 14, thanks to [bun](https://bun.sh/)

Interesting—I just tried it with bun and got exactly the same result as before.

I’ve started cracking open the dependencies to see if I can get more information about the `sass-loader` error.

**Comment by kaelri** [comment]
Followed the thread all the way back to an issue (or set of related issues) with Sass itself: https://github.com/sass/dart-sass/issues/2127

Gonna close this. Temporary downgrade to node@20 is working for me as a workaround. Thanks for everyone’s help. :)

**Comment by kaelri** [comment]
Followed the thread all the way back to an issue (or set of related issues) with Sass itself: https://github.com/sass/dart-sass/issues/2127

Gonna close this. Temporary downgrade to node@20 is working for me as a workaround. Thanks for everyone’s help. :)
